keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,738 @@
+    | grpc ProvisionerCreateBucket({           |
+    |      "name": "br-$uuid",                 |
+    |      "protocol": "s3",                   |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Once the Bucket is successfully created in the OSP, sidecar sets `BucketReady`condition to `True`. At this point, the Bucket is ready to be utilized by workloads.
+
+## Generating Access Credentials for Buckets
 
-When the app pod is started, the kubelet will gRPC call _NodeStageVolume_ and _NodePublishVolume_, which are received by the cosi-node-adapter. The pod waits until the adapter responds to the gRPC requests. The adapter ensures that the target bucket has been provisioned and is ready to be accessed.
+The following stakeholders are involved in the lifecycle of access credential generation:
 
-When the app pod terminates, the kubelet gRPC calls _NodeUnstageVolume_ and _NodeUnpublishVolume_, which the cosi-node-adapter also receives. The adapter orchestrates tear-down of the ephemeral volume provisioned for this bucket.
+ - Users  - request access to buckets
+ - Admins - establish cluster wide access policies
 
-General prep:
-+ admin creates the needed bucket classes and bucket access classes.
+Access credentials are represented by BucketAccess objects. Here are the steps for creating a BucketAccess:
 
-Prep for brownfield:
-+ admin creates `Bucket` instances representing backend buckets.
+  ![COSI Access Credential Creation](images/create-bucket-access.jpg)
 
-## Create Bucket
+###### 1. Admin creates BucketAccessClass, User requests BucketAccess to be created
 
-![CreateBucket Workflow](images/create-bucket.png)
+The BucketAccessClass represents a set of common properties shared by multiple BucketAccesses. It is used to specify policies for creating access credentials, and also for configuring driver-specific access parameters. More information about BucketAccessClass is [here](#bucketclass)
 
-This workflow describes the automation supporting creating a new (greenfield) backend bucket. Accessing this bucket is covered in [Sharing COSI Created Buckets](#sharing-cosi-created-buckets). COSI determines that a request is for an existing bucket when `BucketRequest.bucketInstanceName` is specified.
+The BucketAccessRequest is a request to create BucketAccess. This resource is designed to specify the Bucket for which the credentials provide access, and also for specifying access modes (rw, ro, wo) for that the application requires. More information about BucketAccessRequest is [here](#bucketaccessrequest)
 
-Here is the workflow:
+BucketSecret is the name of the secret that COSI will generate with credentials to access the bucket. The same secret name has to be set in the podSpec to as a projected secret volume.
```

Comment:
A link to the podSpec would be helpful here.
Also, the last sentence is a bit off. Do you mean: "_The same secret name has to be set in the podSpec as a projected secret volume._"